backend/src/services/log.ts

```diff
@@ -76,6 +84,11 @@ export class Writer extends WritableStream {
     return typeof value === 'object' ? util.inspect(value) : String(value)
   }
 
+  static isRequestCompleted(data: any): data is RequestCompletedLog {
```

Comment:
Should be using `unknown` instead of `any`. Looks like this is an issue in a number of places in this file, but I'll refrain from commenting on those.

If `data.msg` is guaranteed then we can at the very least use generics to improve the typing.